server/asset/eth/rpcclient.go

```diff
@@ -175,27 +230,28 @@ func (c *rpcclient) swap(ctx context.Context, assetID uint32, secretHash [32]byt
 // transaction gets the transaction that hashes to hash from the chain or
 // mempool. Errors if tx does not exist.
 func (c *rpcclient) transaction(ctx context.Context, hash common.Hash) (tx *types.Transaction, isMempool bool, err error) {
-	return c.ec.TransactionByHash(ctx, hash)
+	return tx, isMempool, c.withClient(func(ec *ethConn) error {
+		tx, isMempool, err = ec.TransactionByHash(ctx, hash)
+		return err
+	}, true)
```

Comment:
Maybe a trailing comment because this is easy to miss: `}, true) // stop on first provider with "not found", because this shoudl be an error if tx does not exist`